spec/classes/init_spec.rb

```diff
@@ -134,6 +134,27 @@
         end
       end
 
+      describe 'without TLS client authenticatio' do
+        let :params do
+          { use_client_tls_certs: false }
+        end
+        it 'should create puppet.yaml' do
+          verify_exact_contents(catalogue, "#{etc_dir}/foreman.yaml", [
+            "---",
+            ':url: "https://foreman.example.com"',
+            ":ssl_ca: \"#{ssl_dir}/certs/ca.pem\"",
+            ":puppetdir: \"#{var_dir}\"",
+            ':puppetuser: "puppet"',
+            ':facts: true',
+            ":fact_extension: \"#{fact_extension}\"",
+            ':timeout: 60',
+            ':report_timeout: 60',
+            ':report_retry_limit: 1',
+            ':threads: null',
+          ])
+        end
```

Comment:
I think rubocop would want an empty line after let. Then I'd also prefer:
```suggestion

        it { is_expected.to contain_file("#{etc_dir}/foreman.yaml").without_content(%r{:ssl_(cert|key):}) }
```